index.js

```diff
@@ -32,28 +32,72 @@ const fastifyMail = async (fastify, opts) => {
 
   // mail decorator configurations:
   const mail = {
-    createMessage: async function (recipients, templates, context) {
+    // createMessage: async function (recipients, templates, context, opts = {}) {
+    createMessage: async function (message, templatePath, context) {
+      // renders a template with the given context based on the templateName which
+      // should be found in the path provided by templates. Returns "" if the promise is rejected.
+      const renderTemplate = async (templateName) => {
```

Comment:
move supporting functions further down rather than at the head of the main most interesting block (if possible)?

Reply:
done

Reply:
sitting on a commit? are we still going to move this function near the end of the createMessage block?

Reply:
yep done